DTAConfig/CustomSettings/CustomSettingFileCheckBox.cs

```diff
@@ -0,0 +1,161 @@
+                    break;
+
+                string[] parts = fileInfo.Split(',');
+                if (parts.Length < 2)
+                {
+                    Logger.Log($"Invalid CustomSettingFileCheckBox information in {Name}: {fileInfo}");
+                    continue;
+                }
+
+                FileOperationOptions options = default;
+                if (parts.Length >= 3)
+                    Enum.TryParse(parts[2], out options);
+
+                disabledFiles.Add(new FileSourceDestinationInfo(parts[0], parts[1], options));
+
+                i++;
+            }
+        }
+
+        public override void ParseAttributeFromINI(IniFile iniFile, string key, string value)
+        {
+            switch (key)
+            {
+                case "DefaultValue":
+                    defaultValue = Conversions.BooleanFromString(value, false);
+                    return;
+                case "CheckFilePresence":
+                    checkFilePresence = Conversions.BooleanFromString(value, false);
+                    return;
+                case "ResetUnselectableItem":
+                    resetUnselectableItem = Conversions.BooleanFromString(value, false);
+                    return;
+                case "RestartRequired":
+                    restartRequired = Conversions.BooleanFromString(value, false);
+                    return;
+            }
+
+            base.ParseAttributeFromINI(iniFile, key, value);
+        }
+
+        public void Load()
+        {
+            Checked = UserINISettings.Instance.GetCustomSettingValue(Name, defaultValue);
+            originalState = Checked;
+        }
+
+        public bool RefreshSetting()
+        {
+            bool currentValue = Checked;
+
+            if (checkFilePresence)
+            {
+                Enabled = true;
+                
+                if (resetUnselectableItem)
+                {
+                    if (DisabledFilesComplete != EnabledFilesComplete)
+                        Checked = EnabledFilesComplete;
+                    else if (!DisabledFilesComplete && !EnabledFilesComplete)
+                        Checked = defaultValue;
+                }
+            }
+
+            return Checked != currentValue;
+        }
+
+        public bool Save()
+        {
+            bool canBeChecked = !checkFilePresence || EnabledFilesComplete;
+            bool canBeUnchecked = !checkFilePresence || DisabledFilesComplete;
+
+            if (Checked && canBeChecked)
+            {
+                disabledFiles.ForEach(f => f.Revert());
+                enabledFiles.ForEach(f => f.Apply());
+            }
+            else if (!Checked && canBeUnchecked)
+            {
+                enabledFiles.ForEach(f => f.Revert());
+                disabledFiles.ForEach(f => f.Apply());
+            }
+            else // undefined state, the checkbox is bork, don't continue
+            {
+                return false;
```

Comment:
Are there any situations where this could happen? If it happens, should the situation be fixable through the client or is the user left to mess with the files by themself?

This looks like something that should likely be logged.

Reply:
In current implementation this is would be either mod config maker's fault when there would be no valid states/items chosen by `Refresh()` when the option is reset or the installation is broken.